File: fizz-buzz.py
#!/usr/bin/env python

for i in range(1, 100+1):
    def divides(mod):
        return i % mod == 0
    div3 = divides(3)
    div5 = divides(5)

    if div3:
        if div5:
            print("FizzBuzz")
        else:
            print("Fizz")
    elif div5:
        print("Buzz")
    else:
        print(i)

'''
=head1 ABOUT

The FizzBuzz program. See:

=over 4

=back

=head1 COPYRIGHT & LICENSE

Copyright 2017 by Shlomi Fish

This program is distributed under the MIT / Expat License:
L<http://www.opensource.org/licenses/mit-license.php>

Permission is hereby granted, free of charge, to any person
obtaining a copy of this software and associated documentation
files (the "Software"), to deal in the Software without
restriction, including without limitation the rights to use,
copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the
Software is furnished to do so, subject to the following
conditions:

The above copyright notice and this permission notice shall be
included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
OTHER DEALINGS IN THE SOFTWARE.

=cut
'''

File: fizz-buzz.jl
#=
    fizz-buzz
    Copyright © 2019 Shlomi Fish <shlomif@cpan.org>

    Distributed under terms of the MIT license.
=#


for n in 1:100
    m3 = n % 3 == 0
    m5 = n % 5 == 0
    if m3
        if m5
            println("FizzBuzz")
        else
            println("Fizz")
        end
    else
        if m5
            println("Buzz")
        else
            println(n)
        end
    end
end
